### PR TITLE
Add Redis module

### PR DESCRIPTION
**Pull request by djakielski**
Adds new testcontainer for redis

Task:
- [x] docs
- [x] write & read tests
- [x] execute redis cli command
- [x] container start
- [x] password authentication 
- [x] persistent volume

### CODE REVIEW

**Comment by djakielski** [comment]
Hi @cristianrgreco, it's ready for review.

**Comment by djakielski** [comment]
Hi @cristianrgreco, who can I ask for a review? There's no contribution guide lines or a how to.

**Comment by cristianrgreco** [comment]
Hi @djakielski, I've been away the last couple of weeks. I'm back now and will have a look. Thanks for the contributions!

File: packages/modules/redis/src/redis-container.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { AbstractStartedContainer, GenericContainer, StartedTestContainer } from "testcontainers";
+
+const REDIS_PORT = 6379;
+
+export class RedisContainer extends GenericContainer {
+  private password = "";
+  private persistenceVolume = "";
```
Comment (cristianrgreco) [inline packages/modules/redis/src/redis-container.ts:7]: Let's use an optional type `private persistenceVolume?: string` and then check `if (this.persistenceVolume)` to get type-checking support

File: packages/modules/redis/src/redis-container.test.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  it("should execute container cmd and return the result", async () => {
+    const container = await new RedisContainer().start();
+
+    const queryResult = await container.executeCliCmd("info", ["clients"]);
+    expect(queryResult).toEqual(expect.stringContaining("connected_clients:1"));
+
+    await container.stop();
+  });
+  // }
+  // simpleConnect {
```
Comment (cristianrgreco) [inline packages/modules/redis/src/redis-container.test.ts:82]:
```suggestion

  // simpleConnect {
```

File: packages/modules/redis/package.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  "name": "@testcontainers/redis",
+  "version": "10.2.1",
+  "license": "MIT",
+  "keywords": [
+    "redis",
+    "testing",
+    "docker",
+    "testcontainers"
+  ],
+  "description": "redis module for Testcontainers",
```
Comment (cristianrgreco) [inline packages/modules/redis/package.json:11]:
```suggestion
  "description": "Redis module for Testcontainers",
```

**Review by cristianrgreco — COMMENTED** [review]
Only nitpicks - a great contribution, thank you!

I'll need to think of something else to use as the [Quickstart example](https://github.com/testcontainers/testcontainers-node/blob/c97e2942f047814a5d80835402cf2fcf3d944913/docs/quickstart.md?plain=1#L19-L43) 😄

**Comment by djakielski** [comment]
Hi @cristianrgreco, how can i start all these test again? I had fixed an issue and want to retest it.

**Comment by bijela-gora** [comment]
Nice contribution!

File: packages/modules/redis/src/redis-container.test.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    const container = await new RedisContainer().withPassword("test").start();
+
+    const client = await connectTo(container);
+
+    await client.set("key", "val");
+    expect(await client.get("key")).toBe("val");
+
+    await client.disconnect();
+    await container.stop();
+  });
```
Comment (eddumelendez) [inline packages/modules/redis/src/redis-container.test.ts:34]: is it ok to not build the client using the password? 🤔

File: packages/modules/redis/src/redis-container.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+  public override async start(): Promise<StartedRedisContainer> {
+    this.withExposedPorts(...(this.hasExposedPorts ? this.exposedPorts : [REDIS_PORT]))
+      .withCommand([
+        "redis-server",
+        ...(this.password ? [`--requirepass "${this.password}"`] : []),
+        ...(this.persistenceVolume ? ["--save 1 1 ", "--appendonly yes"] : []),
+      ])
+      .withStartupTimeout(120_000);
+    if (this.persistenceVolume) this.withBindMounts([{ mode: "rw", source: this.persistenceVolume, target: "/data" }]);
```
Comment (eddumelendez) [inline packages/modules/redis/src/redis-container.ts:31]: I don't think this is something we would like to do. Testcontainers encourages to use `copyFilesToContainer` instead and make sure the tests are portable across docker environments.

**Review by eddumelendez — COMMENTED** [review]
Thanks for your contribution, @djakielski ! I've left some comments.

File: packages/modules/redis/src/redis-container.test.ts (hunk 4 above)
Comment (djakielski) [reply packages/modules/redis/src/redis-container.test.ts:34]: The password is part of the Connection URL and will be used in the getConnectionUrl function which is used by connectTo.

File: packages/modules/redis/src/redis-container.ts (hunk 5 above)
Comment (djakielski) [reply packages/modules/redis/src/redis-container.ts:31]: I was thinking about persisting Tests. But on the view of making tests independent, you're right. I will fix it.

Comment (cristianrgreco) [reply packages/modules/redis/src/redis-container.ts:31]: @djakielski AFAIK this is the only change pending

Comment (djakielski) [reply packages/modules/redis/src/redis-container.ts:41]: @cristianrgreco I had some trouble with correct way of importing data via redis cli when executing command in the container. I solved it via bash script.

**Comment by djakielski** [comment]
@cristianrgreco do you have any idea why the tests are failing? What did i wrong?

**Comment by cristianrgreco** [comment]
@djakielski I've noticed some intermittent issues with the GHA runners over the last few days. I've kicked off the tests again, hopefully the issue is now resolved. Nothing related with this PR

**Comment by djakielski** [comment]
@cristianrgreco could you rerun the tests? Seems like another temporary issue.

**Comment by cristianrgreco** [comment]
Thanks for your patience and effort @djakielski, much appreciated!

**Pull request merged**